### ISSUE TITLE
Conflict with dependency in project ':app'. Resolved versions for app (15.0.2) and test app (12.0.1) differ  - (onesignal&firebase issue)

### ISSUE THREAD

**Issue by axilaris** [OneSignal/OneSignal-Android-SDK#613]
**Description:**

I cant seem to build gradle for my app on Android Studio 3.0.1 (generate apk works). This used to work without problem until I made a build for another project (which included firebase integration) and then suddenly this does not work. Below was the build output.

```
Error:FAILURE: Build failed with an exception.

What went wrong:
Execution failed for task ':app:preDebugAndroidTestBuild'.
Conflict with dependency 'com.google.firebase:firebase-messaging' in project ':app'. 
Resolved versions for app (15.0.2) and test app (12.0.1) differ. See https://d.android.com/r/tools/test-apk-dependency-conflicts.html for details.

Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

Get more help at https://help.gradle.org

BUILD FAILED in 23s
```

I posted the same question here:
https://stackoverflow.com/questions/51867714/conflict-with-dependency-in-project-app-resolved-versions-for-app-15-0-2-a

**Environment**

1. What version of the Android SDK are you using?
Android Studio 3.0.1 
2. How did you add the SDK to your project (eg. maven)
in build.gradle
implementation 'com.onesignal:OneSignal:[3.9.1, 3.99.99]'
you can see build.gradle code (project and app and error output) here
https://gist.github.com/axilaris/3151780bfddc5efbf8f1928d2874230f

```
dependencies {
    implementation fileTree(dir: 'libs', include: ['*.jar'])
    implementation 'com.android.support:appcompat-v7:26.1.0'
    implementation 'com.android.support.constraint:constraint-layout:1.0.2'
    implementation 'com.android.support:support-v4:26.1.0'
    testImplementation 'junit:junit:4.12'
    androidTestImplementation 'com.android.support.test:runner:1.0.1'
    androidTestImplementation 'com.android.support.test.espresso:espresso-core:3.0.1'
    implementation 'com.squareup.retrofit2:retrofit:2.3.0'
    implementation 'com.squareup.retrofit2:converter-gson:2.3.0'
    implementation 'com.google.code.gson:gson:2.8.1'
    implementation 'com.squareup.okhttp3:okhttp:3.10.0'
    implementation 'com.github.d-max:spots-dialog:0.7@aar'
    //implementation 'com.squareup.retrofit2:converter-moshi:2.0.2'
    implementation 'com.android.support:design:26.1.0'
    implementation 'com.android.support:recyclerview-v7:26.1.0'
    implementation 'com.github.PhilJay:MPAndroidChart:v3.0.3'
    implementation 'com.android.support:cardview-v7:26.1.0'

    implementation 'com.google.android.gms:play-services-places:15.0.1'
    implementation 'com.google.android.gms:play-services-maps:15.0.1'

    implementation 'com.onesignal:OneSignal:[3.9.1, 3.99.99]'


    implementation('com.crashlytics.sdk.android:crashlytics:2.9.4@aar') {
        transitive = true;
    }

}
```

**Steps to Reproduce Issue:**

  1. This happens after building for another project (which may have included firebase&gradle upgrades). initially the same environment can build gradle without any problems and generate apk
(on another android studio, it works, just not on my environment)

**Anything else:**

you can see build.gradle code (project and app and error output) here
https://gist.github.com/axilaris/4053a5447692484e660907cacf2274ce

I tried to refer to these 2 very similar build issues but unable to resolve it. 
https://github.com/OneSignal/OneSignal-Android-SDK/issues/327
https://github.com/OneSignal/OneSignal-Android-SDK/issues/609

i tried buildToolsVersion '25.0.3' but it has then this ignore
Warning:The specified Android SDK Build Tools version (26.0.2) is ignored, as it is below the minimum supported version (27.0.3) for Android Gradle Plugin 3.1.0.

**Comment by jkasten2** [comment]
@axilaris Thanks for all the details and sharing your `build.gradle` files. However I am not able to reproduce the issue. Can you try cleaning your project? You can do so from Android Studio or with `./gradlew clean` from the terminal.

If you still have the issue can you share the `./gradlew` command you ran to reproduce the error? Such as `./gradlew build`? Or if building through Android studio did you just do standard build?

Can you also try updating to the latest 3.1.4 Android Studio?

Lastly if you can provide a full example project can you reproducing the issue?

**Comment by jkasten2** [comment]
Closing due to inactivity, please provide more details and we can look into the issue further.

**Comment by aleesha711** [comment]
I am having same issue
Conflict with dependency 'com.google.firebase:firebase-messaging' in project ':app'. Resolved versions for app (15.0.2) and test app (12.0.1)

**Comment by jkasten2** [comment]
@aleesha711 Can you provide the same details as I noted above? If you can provide an minimum example project this will help us reproduce the issue quicker. Thanks.
